src/Middleware/SocialAuthMiddleware.php

```diff
@@ -176,11 +183,13 @@ protected function _handleCallbackAction(ServerRequest $request, Response $respo
 
         $user->unsetProperty($config['fields']['password']);
 
+        $request->session()->write($config['sessionKey'], $user);
+
         if (!$config['userEntity']) {
             $user = $user->toArray();
         }
 
-        $request->session()->write($config['sessionKey'], $user);
```

Comment:
This commit was reverted in d9598f31933521fa11b2df3af108994dce6f68b9 as requested.